### ISSUE TITLE
Investigate Rider support

### ISSUE THREAD

**Issue by mvonballmo** [mvonballmo/APE_Apps#1]

**Comment by mvonballmo** [comment]
- Maui with Rider
- Xamarin with Rider

**Comment by BigGold1310** [comment]
MAUI Support: https://docs.microsoft.com/en-us/dotnet/maui/supported-platforms
It doesn't support Linux systems at all.
Xamarin also doesn't support Linux. There where a few workarounds around but these change all few months/weeks as the SDK updates.
JetBrains also looked into one but gave up at some point to update it all the time: https://rider-support.jetbrains.com/hc/en-us/articles/360000557259--Obsolete-How-to-develop-Xamarin-Android-applications-on-Linux-with-Rider

**Comment by mvonballmo** [comment]
Maui doesn't support Linux _as a compilation target_. That's different from not being able to use Linux to develop Maui apps. I understand why no-one would be prioritizing this because many of the Maui targets are not available on Linux for _other_ reasons:

- MacOS Catalyst? Need MacOS.
- iOS? Need MacOS.
- WinUI? Need Windows.
- Android? _MAYBE_ ... if the emulator works or if the connection to an Android phone works

The MSDN page does mention that,

> Linux, supported by the community

is also a target.

Microsoft does not support Linux as an IDE host. JetBrains Rider does, but it seems to be struggling to [allocate resources to Xamarin development](https://youtrack.jetbrains.com/issue/RIDER-5747) (that issue was linked by JetBrains from the issue you linked above).

It looks like the article [Rider Xamarin.Android](https://github.com/Flying--Dutchman/RiderXamarinAndroid/blob/main/README.md) got it running on Arch Linux. See the [user's comment](https://rider-support.jetbrains.com/hc/en-us/articles/360000557259/comments/4405028734866).

I've posted comments on both the open issue for Xamarin above and also on the recently published article [macOS Environment Setup For MAUI Development](https://blog.jetbrains.com/dotnet/2022/05/25/macos-environment-setup-for-maui-development/). That setup worked great for me on MacOS.

**Comment by mvonballmo** [comment]
Just got a [response from Rodion Iafarov of JetBrains](https://youtrack.jetbrains.com/issue/RIDER-5747/Add-XamarinAndroid-support-on-Linux#focus=Comments-27-6141521.0-0):

"With MAUI things are better, as there is no need to package Xamarin SDK. As of now MAUI is not supported on linux, whereas maui-android workload can be installed already now. As we have added basic support for MAUI for run and debug, it actually already works on linux for android native apps.

It's still to be evaluated if anything has to be done on top, as linux has lower priority, but in case dotnet stack will work same way, it's very likely that Rider will provide same functionality on linux. In short, we cannot confirm any development plans for MAUI support in Rider on linux yet, but many parts can just work."

Basically, it sounds like you can try it? They haven't actually done testing, but they also see no reason why it shouldn't work. Good luck?!??
